doc/dev-guide/src/linting.md

```diff
@@ -0,0 +1,33 @@
+# Linting
+
+## Manual Linting
+
+When checking the codebase with [`Clippy`](https://doc.rust-lang.org/stable/clippy/index.html), one needs to use:
+
+`cargo +beta clippy --all --all-targets --all-features -- -D warnings`
+
+## Rust-Analyzer
+
+When using  [`Rust-Analyzer`](https://rust-analyzer.github.io/) integration in the IDE of your choice so that you can check lints live while editing the codebase, in order for it to correctly work its necessary to configure `rust-analyzer` settings. 
+
+You will need to set the `rust-analyzer.cargo.features` configuration to `"all"` (Check the [`rust-analyzer` manual](https://rust-analyzer.github.io/manual.html) for further detail on how to configure it).
+
+### VsCode/VsCodium setup
+
+Add 
+
+```json
+"rust-analyzer.cargo.features": "all":,
+```
+
+in your project at `.vscode/settings.json`
+
+or
+
+to your global configuration `~/.config/Code/User/settings.json` (although you need to be aware that this will apply to all your Rust projects)
```

Comment:
```suggestion
to your global configuration `~/.config/Code/User/settings.json` (although you need to be aware that this will apply to all your Rust projects).
```